tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -503,6 +503,176 @@ def test_polyroutine_diff_types_poly1d(self, xp, dtype1, dtype2):
+
+    @testing.for_all_dtypes(no_float16=True)
+    def test_polyfit_y_invalid_shape(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((5,), xp, dtype)
+            y = testing.shaped_arange(self.shape, xp, dtype)
+            with pytest.raises(TypeError):
+                xp.polyfit(x, y, 5)
+
+    @testing.for_all_dtypes(no_float16=True)
+    def test_polyfit_w_invalid_shape(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((5,), xp, dtype)
+            w = testing.shaped_arange(self.shape, xp, dtype)
+            with pytest.raises(TypeError):
+                xp.polyfit(x, x, 5, w=w)
+
+
+@testing.gpu
+class TestPolyfitInvalid(unittest.TestCase):
+
+    @testing.for_all_dtypes(no_float16=True)
+    def test_polyfit_neg_degree(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((5,), xp, dtype)
+            y = testing.shaped_arange((5,), xp, dtype)
+            with pytest.raises(ValueError):
+                xp.polyfit(x, y, -4)
+
+    @testing.for_all_dtypes(no_float16=True)
+    def test_polyfit_complex_degree(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((5,), xp, dtype)
+            y = testing.shaped_arange((5,), xp, dtype)
+            with pytest.raises(TypeError):
+                xp.polyfit(x, y, 5j)
+
+    @testing.for_all_dtypes(no_float16=True)
+    def test_polyfit_xy_mismatched_length(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((10,), xp, dtype)
+            y = testing.shaped_arange((5,), xp, dtype)
+            with pytest.raises(TypeError):
+                xp.polyfit(x, y, 5)
+
+    @testing.for_all_dtypes(no_float16=True)
+    def test_polyfit_yw_mismatched_length(self, dtype):
+        for xp in (numpy, cupy):
+            y = testing.shaped_arange((10,), xp, dtype)
+            w = testing.shaped_arange((5,), xp, dtype)
+            with pytest.raises(TypeError):
+                xp.polyfit(y, y, 5, w=w)
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True)
+    def test_polyfit_cov_invalid(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((5,), xp, dtype)
+            y = testing.shaped_arange((5,), xp, dtype)
+            with pytest.raises(ValueError):
+                xp.polyfit(x, y, 5, cov=True)
+
+
+@testing.gpu
+class TestPolyfitDiffTypes(unittest.TestCase):
+
+    @testing.for_all_dtypes_combination(
+        names=['dtype1', 'dtype2'], no_float16=True, no_bool=True)
+    @testing.numpy_cupy_allclose()
+    def test_polyfit_unweighted_diff_types(self, xp, dtype1, dtype2):
+        x = testing.shaped_arange((5,), xp, dtype1)
+        y = testing.shaped_arange((5,), xp, dtype2)
+        return xp.polyfit(x, y, 5)
+
+    @testing.for_all_dtypes_combination(
+        names=['dtype1', 'dtype2', 'dtype3'],
+        no_float16=True, no_bool=True, no_complex=True)
```

Comment:
```suggestion
        no_float16=True)
```

Reply:
I forgot to remove `no_bool=True` in dtypes combinations tests. Thanks!
Both numpy and cupy raise Exception when ` no_complex=True` is removed because of `lhs *= w[:, None]` and I don't think weights should be allowed to be complex in general.

Reply:
How about checking if the same type of error is raised by passing `accept_error` option to `testing.numpy_cupy_allclose`?